First implementation of Orphanage farm

maxux opened the pull request.

maxux committed:
First implementaton of Orphanage farm

maxux committed:
Merge branch 'master' into master-default-farm

zaibon commented on tools/tffarmer/main.go:
```diff
@@ -35,7 +35,7 @@ func main() {
 			Name:   "tnodb, u",
 			Usage:  "URL of the TNODB",
-			Value:  "https://tnodb.dev.grid.tf",
+			Value:  "https://bcdb.dev.grid.tf",
```
you can also rename the flag to `bcdb`

zaibon commented:
We should also prevent provisiond to deploy when there is no farmer_id defined.

maxux commented:
Maybe not prevent everything, it's cool to keep `debug` for example ?

zaibon commented:
> Maybe not prevent everything, it's cool to keep debug for example ?

I don't really see the point. If there is no farmer_id, then the node should not be used for anything. So there wouldn't be anything to debug anyhow.

maxux committed:
Update provisiond and identityd to use environment orphan

The environment manager take care to fetch the farmer id or set farmer
id according th orphanage.

If the node is orphan, provisiond doesn't start.

zaibon commented on cmds/provisiond/main.go:
```diff
@@ -56,6 +56,14 @@ func main() {
 		log.Fatal().Msgf("fail to connect to message broker server: %v", err)
 	}
 
+	env := environment.Get()
```
it looks like you can do this even before creating the redis client at line 54

maxux committed:
Check for orphan on provisiond before connecting redis

maxux committed:
Merge branch 'master' into master-default-farm
